content/ytt/docs/latest/data-values-schema-migration-guide.md

```diff
@@ -0,0 +1,201 @@
+  port: 8080
+```
+You can now copy this output into a yaml file and make it a schema by adding the `#@data/values-schema` annotation to the top of a new document.
+
+`values-schema.yml`:
+```yaml
+#@data/values-schema
+---
+key1: myVal
+key2: 8088
+key3:
+  host: registry.dev.io
+  port: 8080
+```
+
+Now just include the `values-schema.yml` file in your `ytt` invocation instead of the multiple data values files.
+
+### Multiple data values files + Private libraries
+In this scenario, we expect your configuration to depend on a ytt library; outlined in the [library module docs](lang-ref-ytt-library.md).
+
+```bash
+$ tree .
+.
+└── config
+    ├── config.yml 
+    ├── values-1.yml
+    ├── values-2.yml
+    └── _ytt_lib
+        └── lib
+            ├── service.yml
+            └── values.yml
+```
+`config.yml`:
+```yaml
+#@ load("@ytt:data", "data")
+#@ load("@ytt:library", "library")
+#@ load("@ytt:template", "template")
+
+#@ lib = library.get("lib").with_data_values(data.values)
+--- #@ template.replace(lib.eval())
+```
+Using the same `values-1.yml` and `values-2.yml` files from the [multiple data values files schema migration example above](#multiple-data-values-files).
+Migrating to schema happens one library at a time. Let's start with the root library, which includes everything at and below the file system level where the `ytt` invocation was called, not including the `_ytt_lib` folder:
+```bash
+.
+└── config
+    ├── config.yml
+    ├── values-1.yml
+    └── values-2.yml
+```
+As seen in the [previous example](#multiple-data-values-files), `values-1.yml` and `values-2.yml` can be combined and replaced with a single schema file:
+`values-schema.yml`:
+```yaml
+#@data/values-schema
+---
+key1: myVal
+key2: 8088
+key3:
+  host: registry.dev.io
+  port: 8080
+```
+Now we have migrated the root library to use schemas, and the `ytt` invocation will succeed as the same as before, but the private library, "lib" is not yet migrated to schema. 
+```bash
+$ tree .
+.
+└── config
+    ├── config.yml 
+    ├── values-schema.yml
+    └── _ytt_lib
+        └── lib
+            ├── service.yml
+            └── values.yml
+```
+Migrating a private library to use schemas involves the same process as the root library. You can narrow the context to just the children of the `_ytt_lib` directory:
+```bash
+└── _ytt_lib
+    └── lib
+        ├── service.yml
+        └── values.yml
+```
+Now simply follow the steps in either of the previous examples to migrate the private library to use schemas.
+
+## How do I provide default values for an array?
+Arrays in schemas have [special behavior](lang-ref-ytt-schema.md#inferring-defaults-for-arrays), their default value is empty, but their inferred type is from the value in the schema document. 
+Arrays must have one and only one item, which provides the inferred type of that data value. The example below shows how to define an array in a schema and them provide default values via a data values file.
+
```

Comment:
This is a tricky set of facts to unfurl. 😅 

In a somewhat complex situation — like this one — getting nit-picky precise about what everything is can help avoid confusion.

- _pedantically_, it's the type of the _elements_ that's being declared.
- "inferred" is slightly more accurate as a verb than an adjective, here.
- lead in with the fact that only one value is specified for arrays — this makes it possible to "chronologically" connect this set of related facts.

```suggestion
Arrays in schemas have [special behavior](lang-ref-ytt-schema.md#inferring-defaults-for-arrays): exactly one element is specified in the schema of an array, and that value is _only_ used to infer the type of that array's elements — the default value for arrays is always empty.

The example below shows how to define an array in a schema and then provide default values via a data values file.
```

👍🏻  The link to the section in schema reference page is _**critical**_ here. Great pointer.